### PR TITLE
First pass at adding a contributing section

### PR DESCRIPTION
**Pull request by MattMSumner**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -381,6 +381,55 @@ const hubspot = new Hubspot(params)
...
 
+## Contributing
+
+Before submitting a pull request, please make sure the following is done:
+
+1. Fork the repository and create your branch from master.
+1. Run npm install in the repository root.
+1. If you’ve fixed a bug or added code that should be tested, add tests!
+1. Ensure the test suite passes (npm test). Tip: `npm run mocha -- --grep "test name"` is helpful in development.
+1. Format your code with prettier and eslint (npm run lint).
```
Comment (chrisbaldauf) [inline README.md]: there are a few different commands mentioned in passing here, would it be helpful to have one command to rule them all formatted as a code block?

`npm run build` or `npm run preflight` or whatever the current node standard is.

File: README.md [hunk 2]
```diff
@@ -381,6 +381,55 @@ const hubspot = new Hubspot(params)
...
+out requests to HubSpot's api using [nock]. We then recommend running the test
+suite without mocking api requests with `NOCK_OFF=true npm run mocha`. This
+should fail as you'll need some [environment variables] for real requests to
+pass.
+
+[nock]: https://github.com/nock/nock
+[environment variables]: https://github.com/motdotla/dotenv
+
+If you haven't already, create a [developer account] on hubspot. You'll want to
+[create an app] as well. Then, create a new file, `.env` in the root of the
```
Comment (chrisbaldauf) [inline README.md]: I've found a `.env.template` helpful in past projects. Then you can have the environment variable keys (but no values) present and add `.env` to `.gitignore`.

File: README.md [hunk 3]
```diff
@@ -381,6 +381,55 @@ const hubspot = new Hubspot(params)
...
+NOTE: Your HubSpot user ID; This can be found in the same place as your
+Developer HAPIkey in your Developer portal.
+
+[developer account]: https://developer.hubspot.com
+[create a app]: https://developers.hubspot.com/docs/faq/how-do-i-create-an-app-in-hubspot
+[app id]: https://developers.hubspot.com/docs/faq/how-do-i-find-the-app-id
+[oauth access token]: https://developers.hubspot.com/docs/methods/oauth2/oauth2-overview
+
+```.env
+API_KEY="some-key"
```
Comment (chrisbaldauf) [inline README.md]: Can we make these values more representative of the shape of real data? I'd prefer they not be valid, but have the same format.

Comment (MattMSumner) [reply README.md]: Done in 
https://github.com/MadKudu/node-hubspot/pull/138/commits/79d96dc9dfd68f98b68c89a30d2442dd0fe55244

File: README.md (hunk 2 above)
Comment (MattMSumner) [reply README.md]: Added in 
https://github.com/MadKudu/node-hubspot/pull/138/commits/945bf27685ddb34f88f5a6a8b48944a20da00320

File: README.md [hunk 4]
```diff
@@ -396,29 +396,69 @@ MIT
...
+should fail as you'll need some [environment variables] for real requests to
+pass.
+
+[nock]: https://github.com/nock/nock
+[environment variables]: https://github.com/motdotla/dotenv
+
+If you haven't already, create a [developer account] on hubspot. You'll want to
+[create an app] and a [test account] as well. Then, create a new file, `.env` in
+the root of the repo. Inside you'll need to add an [app id], a HubSpot user id,
+an [api key] and an [oauth access token];
```
Comment (SViccari) [inline README.md:428]: > an [api key] and an [oauth access token]

Should this be an "or"? I believe only one means of authentication is necessary.

> an [api key] ~and~ or an [oauth access token]

Comment (MattMSumner) [reply README.md:428]: Unfortunately not. For the test suite to pass you need both as we have tests that use both methods to authenticate.

File: README.md (hunk 1 above)
Comment (MattMSumner) [reply README.md]: Searching GitHub:

search                                                                    | result count
------------------------------------------------------------------------- | ------------
filename:package.json scripts preflight node extension:json language:JSON | 116
filename:package.json scripts build node extension:json language:JSON     | 376,446
filename:package.json scripts setup node extension:json language:JSON     | 61,568

So I'll go with build. Thanks 👍

Comment (MattMSumner) [reply README.md]: Done in 
https://github.com/MadKudu/node-hubspot/pull/138/commits/23e467b0b7988439d1a1835c07fa3b4893327127

**Pull request merged**